pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/invertedindex/RangeIndexHandler.java

```diff
@@ -69,22 +69,31 @@ public boolean needUpdateIndices(SegmentDirectory.Reader segmentReader) {
     String segmentName = _segmentDirectory.getSegmentMetadata().getName();
     Set<String> columnsToAddIdx = new HashSet<>(_columnsToAddIdx);
     Set<String> existingColumns = segmentReader.toSegmentDirectory().getColumnsWithIndex(StandardIndexes.range());
+
+    // Check if any index updates are required.
+    boolean rangeIndexUpdated = false;
+
     // Check if any existing index need to be removed.
     for (String column : existingColumns) {
       if (!columnsToAddIdx.remove(column)) {
         LOGGER.info("Need to remove existing range index from segment: {}, column: {}", segmentName, column);
-        return true;
+        rangeIndexUpdated = true;
       }
     }
-    // Check if any new index need to be added.
     for (String column : columnsToAddIdx) {
       ColumnMetadata columnMetadata = _segmentDirectory.getSegmentMetadata().getColumnMetadataFor(column);
-      if (shouldCreateRangeIndex(columnMetadata)) {
+      if (columnMetadata == null) {
+        continue;
+      }
```

Comment:
It's not quite clear when we'd hit this case but I've retained the existing check.